### PR TITLE
Docs for registering transform_feature_names

### PR DESCRIPTION
**Pull request by jnothman**
WDYT, @kmike?

### CODE REVIEW

File: docs/source/libraries/sklearn.rst [hunk 1]
```diff
@@ -209,6 +209,42 @@ long as:
...
+        def fit(self, X, y=None):
+            # we store n_features_ for the sake of transform_feature_names
+            # when in_names=None:
+            self.n_features_ = check_array(X).shape[1]
+            return self
+
+        def transform(self, X):
+            return check_array(X)[:, 1::2]
+
+    @transform_feature_names.register
```
Comment (kmike) [inline docs/source/libraries/sklearn.rst]: .register(OddTransformer)?

File: docs/source/libraries/sklearn.rst [hunk 2]
```diff
@@ -209,6 +209,42 @@ long as:
...
+    def odd_feature_names(transformer, in_names=None):
+        if in_names is None:
+            from eli5.sklearn.utils import get_feature_names
+            # generate default feature names
+            in_names = get_feature_names(num_features=transformer.n_features_)
```
Comment (kmike) [inline docs/source/libraries/sklearn.rst]: It won't work as-is because currently get_feature_names requires `clf` argument; it seems that's fine to make it optional.

**Comment by jnothman** [comment]
So you can tell I didn't test.

**Comment by kmike** [comment]
Looks great, thanks!
